spec/models/notification_spec.rb

```diff
@@ -519,17 +519,12 @@
   end
 
   describe "#fast_destroy_old_notifications" do
-    let_it_be(:notifications) { create_list(:notification, 3) }
-
-    it "bulk deletes notifications older than 4 months by default" do
-      notifications.last.update(created_at: 5.months.ago)
-      expect { described_class.fast_destroy_old_notifications }.to change(described_class, :count).by(-1)
-    end
-
-    it "bulk deletes notifications older than a given timestamp" do
-      expect do
-        described_class.fast_destroy_old_notifications(Time.current)
-      end.to change(described_class, :count).by(-notifications.size)
+    it "bulk deletes notifications older than given timestamp" do
+      bulk_deleter = BulkSqlDelete.new
+      allow(BulkSqlDelete).to receive(:new).and_return(bulk_deleter)
+      allow(bulk_deleter).to receive(:delete_in_batches)
+      described_class.fast_destroy_old_notifications("a_time")
+      expect(bulk_deleter).to have_received(:delete_in_batches).with(a_string_including("< 'a_time'"))
```

Comment:
hahaha here you're cheating a bit, i can see you're trying to avoid timezone issues :P